composer/utils/file_helpers.py

```diff
@@ -166,7 +169,8 @@ def ensure_folder_has_no_conflicting_files(folder_name: Union[str, pathlib.Path]
 
 
 def format_name_with_dist(format_str: str, run_name: str, **extra_format_kwargs: object):  # noqa: D103
-    formatted_str = format_str.format(
+    formatted_str = partial_format(
```

Comment:
Needed to fix the HF checkpointer, which is implemented and initialized in foundry using the original template string `save_folder`, and it formats its checkpoints using `format_name_with_dist` and `format_name_with_dist_and_time`.. There isn't really a clean way to go back and update its save folder since it's in the form of a generic callback once it reaches composer, but the object store supports placeholders in the upload/download paths.